src/veins/base/phyLayer/MacToPhyControlInfo.h

```diff
@@ -13,99 +12,30 @@ namespace Veins {
-    {
-    }
-
-    /**
-     * @brief Delete the signal if it is still in our ownership.
-     */
-    ~MacToPhyControlInfo() override
-    {
-        if (signal) delete signal;
-    }
-
-    /**
-     * @brief Sets the signal of this MacToPhyControlInfo.
-     *
-     * NOTE: Once a signal is passed to the MacToPhyControlInfo,
-     *          MacToPhyControlInfo takes the ownership of the Signal.
-     */
-    void setSignal(Signal* s)
-    {
-        if (signal) delete signal;
-
-        signal = s;
-    }
-
-    /**
-     * @brief Returns a pointer to the Signal of this ControlInfo.
-     *
-     * NOTE: The ownership of the Signal is passed together with
-     *          the Signal itself. This means the caller of this
-     *          method is responsible for deletion of the Signal.
-     *          MacToPhyControlInfo also loses its pointer to the
-     *          Signal when this Method is called so following
-     *          calls of this method will return null!
-     */
-    Signal* retrieveSignal()
-    {
-        Signal* tmp = signal;
-        signal = nullptr;
-        return tmp;
-    }
-
-    /**
-     * @brief Attaches a "control info" structure (object) to the message pMsg.
-     *
-     * This is most useful when passing packets between protocol layers
-     * of a protocol stack, the control info will contain the signal.
-     *
-     * The "control info" object will be deleted when the message is deleted.
-     * Only one "control info" structure can be attached (the second
-     * setL3ToL2ControlInfo() call throws an error).
-     *
-     * @param pMsg        The message where the "control info" shall be attached.
-     * @param pSignal    The signal which should be send.
-     */
-    static cObject* const setControlInfo(cMessage* const pMsg, Signal* const pSignal)
-    {
-        MacToPhyControlInfo* const cCtrlInfo = new MacToPhyControlInfo(pSignal);
-        pMsg->setControlInfo(cCtrlInfo);
-
-        return cCtrlInfo;
-    }
-    /**
-     * @brief extracts the signal from message "control info".
-     */
-    static Signal* const getSignal(cMessage* const pMsg)
-    {
-        return getSignalFromControlInfo(pMsg->getControlInfo());
-    }
-    /**
-     * @brief extracts the signal from "control info".
-     */
-    static Signal* const getSignalFromControlInfo(cObject* const pCtrlInfo)
-    {
-        MacToPhyControlInfo* const cCtrlInfo = dynamic_cast<MacToPhyControlInfo* const>(pCtrlInfo);
-
-        if (cCtrlInfo) return cCtrlInfo->retrieveSignal();
-        return nullptr;
-    }
+struct VEINS_API MacToPhyControlInfo : public cObject {
+    Spectrum spectrum;
+    double centerFrequency;
+    // FIXME: remove members above this line
+    /** @brief Absolute start time of the frame transmission.*/
+    simtime_t startTime;
+    /** @brief Duration of the frame transmission.*/
+    simtime_t duration;
+    /** @brief Channel number/index used to select frequency (protocol-specific).*/
+    int channelNr;
```

Comment:
as channelNr (and many other fields) are protocol specific, shouldn't they rather be moved to a 802.11p specific subclass?